From 8b6d1fc4ca2825855e6c5a0c281a927c3e887599 Mon Sep 17 00:00:00 2001
From: Sriram R <srirrama@codeaurora.org>
Date: Wed, 27 Jan 2021 12:25:29 +0530
Subject: [PATCH] ath11k: Add counter for packets received on wbm release route

  Add a counter and update the soc_dp stats for packets
received on wbm release ring with reason as 'routing'

Signed-off-by: Sriram R <srirrama@codeaurora.org>
---
 drivers/net/wireless/ath/ath11k/core.h  | 1 +
 drivers/net/wireless/ath/ath11k/debug.c | 2 ++
 drivers/net/wireless/ath/ath11k/dp_rx.c | 1 +
 3 files changed, 4 insertions(+)

diff --git a/drivers/net/wireless/ath/ath11k/core.h b/drivers/net/wireless/ath/ath11k/core.h
index 0d8cbab..eaeeaac 100644
--- a/drivers/net/wireless/ath/ath11k/core.h
+++ b/drivers/net/wireless/ath/ath11k/core.h
@@ -777,6 +777,7 @@ struct ath11k_soc_dp_stats {
 	u32 rxdma_error[HAL_REO_ENTR_RING_RXDMA_ECODE_MAX];
 	u32 reo_error[HAL_REO_DEST_RING_ERROR_CODE_MAX];
 	u32 hal_reo_error[DP_REO_DST_RING_MAX];
+	u32 hal_wbm_rel_route;
 	struct ath11k_soc_dp_tx_err_stats tx_err;
 	struct ath11k_dp_ring_bp_stats bp_stats;
 };
diff --git a/drivers/net/wireless/ath/ath11k/debug.c b/drivers/net/wireless/ath/ath11k/debug.c
index 8ea4132..cfa2dab 100644
--- a/drivers/net/wireless/ath/ath11k/debug.c
+++ b/drivers/net/wireless/ath/ath11k/debug.c
@@ -1463,6 +1463,8 @@ static ssize_t ath11k_debug_dump_soc_dp_stats(struct file *file,
 	len += scnprintf(buf + len, size - len, "SOC RX STATS:\n\n");
 	len += scnprintf(buf + len, size - len, "err ring pkts: %u\n",
 			 soc_stats->err_ring_pkts);
+	len += scnprintf(buf + len, size - len, "wbm rel route pkts: %u\n",
+			 soc_stats->hal_wbm_rel_route);
 	len += scnprintf(buf + len, size - len, "Invalid RBM: %u\n\n",
 			 soc_stats->invalid_rbm);
 	len += scnprintf(buf + len, size - len, "RXDMA errors:\n");
diff --git a/drivers/net/wireless/ath/ath11k/dp_rx.c b/drivers/net/wireless/ath/ath11k/dp_rx.c
index f0c985d..6e027df 100644
--- a/drivers/net/wireless/ath/ath11k/dp_rx.c
+++ b/drivers/net/wireless/ath/ath11k/dp_rx.c
@@ -4732,6 +4732,7 @@ int ath11k_dp_rx_process_wbm_err(struct ath11k_base *ab,
 
 		if (err_info.push_reason !=
 		    HAL_REO_DEST_RING_PUSH_REASON_ERR_DETECTED) {
+			ab->soc_stats.hal_wbm_rel_route++;
 			dev_kfree_skb_any(msdu);
 			continue;
 		}
-- 
2.7.4

